pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataTopicTailer.java

```diff
@@ -69,28 +106,13 @@ public void processRequest(Message<byte[]> msg) {
             serviceRequest = ServiceRequest.parseFrom(msg.getData());
         } catch (IOException e) {
             log.error("Received bad service request at message {}", msg.getMessageId(), e);
-            // TODO: find a better way to handle bad request
-            throw new RuntimeException(e);
+            errorNotifier.triggerError(e);
```

Comment:
Should we just remove the try-catch and let the method throw an exception? Everything is caught in the thread anyways.

Reply:
We would still need to differentiate Interrupted vs others. And if we do encounter some exception isn;'t it better to trigger the errorNotifier and exit perhaps?

Reply:
> We would still need to differentiate Interrupted vs others.

I don't quite follow, the code already differentiates interrupted exceptions vs other exceptions.

Triggering the error in the catch all block in the thread allows us to exit the thread as well and stop further processing of requests.  While this doesn't hurt but is cleaner in a sense.

Reply:
I misunderstood your comments. Yes, that makes sense. Fixed